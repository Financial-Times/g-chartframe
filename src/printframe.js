import frame from './chartframe';
import watermarkPath from './watermarks';

function printFrame(configObject){
  var f = frame()
    .autoPosition(true)
    .containerClass('ft-printgraphic')
    .backgroundColour('#FFF')
    .goalposts('#000')
    .units('mm')
    .width(112.25) //these are after the units are set so they are converted from mm to px
    .height(68)
    .margin({top:40, left:15, bottom:35, right:7})
    .watermark(watermarkPath.dark)
    .rem(9.6)
    .titleStyle({
      'font-size': '12px',
      'fill': '#000000',
      'font-weight': '600',
      'font-family': 'MetricWeb,sans-serif',
    })
    .titleX(7)
    .titleY(15)
    .titleLineHeight(13)
    .subtitleStyle({
      'fill': '#000000',
      'font-size': '9.6px',
      'font-weight': 400,
      'font-family': 'MetricWeb,sans-serif',
    })
    .subtitleLineHeight(10)
    .subtitleX(7)
    .subtitleY(27)
    .sourceStyle({
        'fill': '#000000',
        'font-size': '7.2px',
        'font-weight': 400,
      'font-family': 'MetricWeb,sans-serif',
    })
    .sourceX(7)
    .sourcePlotYOffset(18)
    .sourceLineHeight(8)
    .watermark('');

    if(configObject !== undefined) f.attrs(configObject);
    return f;
}

export default printFrame;
