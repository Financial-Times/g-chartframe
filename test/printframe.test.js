import * as fs from 'fs';
import * as d3 from 'd3-selection';
import jsdom from 'jsdom';
import printframe from '../src/printframe';

test('printframe renders as expected with defaults', () => {
    const { JSDOM } = jsdom;
    const defaultFrame = printframe();
    const dom = new JSDOM(fs.readFileSync('test/scaffold.html'));
    const chartContainer = d3.select(dom.window.document.querySelector('svg'));
    chartContainer.call(defaultFrame);
    expect(chartContainer.node()).toMatchSnapshot();
});

test('printframe with constructor options', () => {
    const frame = printframe({ title: 'PrintFrame' });

    expect(frame.title()).toBe('PrintFrame');
});
