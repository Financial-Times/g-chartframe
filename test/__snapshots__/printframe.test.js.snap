// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`webframe-s renders as expected with defaults 1`] = `
<svg
  aria-labelledby="ft-printgraphic__chart-a11y-title ft-printgraphic__chart-a11y-desc"
  class="ft-printgraphic"
  font-family="MetricWeb,sans-serif"
  role="img"
  xmlns="http://www.w3.org/2000/svg"
>
  <title
    id="ft-printgraphic__chart-a11y-title"
  >
    A chart
  </title>
  <desc
    id="ft-printgraphic__chart-a11y-desc"
  >
    A graphic by the Financial Times
  </desc>
  <rect
    class="chart-background"
    id="chart-background"
    role="presentation"
  />
  <path
    class="chart-goalposts"
  />
  <path
    class="chart-goalposts"
  />
  <text
    class="chart-title"
    id="ft-printgraphictitle"
  >
    <tspan
      fill="#000000"
      font-family="MetricWeb,sans-serif"
      font-size="12px"
      font-weight="600"
      x="7"
      y="15"
    >
      Title: A description of the charts purpose
    </tspan>
  </text>
  <text
    class="chart-subtitle"
    id="ft-printgraphicsubtitle"
  >
    <tspan
      fill="#000000"
      font-family="MetricWeb,sans-serif"
      font-size="9.6px"
      font-weight="400"
      id="ft-printgraphicsubtitle"
      x="7"
      y="27"
    >
      some supporting information, units perhaps
    </tspan>
  </text>
  <text
    class="chart-source"
    id="ft-printgraphicsource"
  >
    <tspan
      fill="#000000"
      font-family="MetricWeb,sans-serif"
      font-size="9.6px"
      font-weight="400"
      id="ft-printgraphicsource"
      x="7"
      y="187.75584678001326"
    >
      Source: research
    </tspan>
  </text>
  <g
    class="chart-watermark"
    role="presentation"
    transform="translate(154.1888794273013,182.75584678001326) scale(1) "
  />
  <g
    class="chart-plot"
    role="presentation"
    transform="translate(15,41.2)"
  />
</svg>
`;
